plasmapy/dispersion/two_fluid_dispersion.py

```diff
@@ -0,0 +1,273 @@
+__all__ = ["two_fluid_dispersion_solution", "tfds_"]
+
+import astropy.units as u
+import numpy as np
+
+from astropy.constants.si import c, e, k_B, m_e, m_p, mu0
+
+import plasmapy.formulary.parameters as pfp
+
+from plasmapy.utils.decorators import validate_quantities
+
+
+@validate_quantities(
+    B={"can_be_negative": False},
+    m_e={"can_be_negative": False},
+    m_i={"can_be_negative": False},
+    n={"can_be_negative": False},
+    T_e={"can_be_negative": False, "equivalencies": u.temperature_energy()},
+    T_i={"can_be_negative": False, "equivalencies": u.temperature_energy()},
+    theta={"can_be_negative": True},
+)
+def two_fluid_dispersion_solution(
+    *,
+    B: u.T,
+    k: u.m ** -1,
+    n: u.m ** -3,
+    T_e: u.K,
+    T_i: u.K,
+    gamma_e=1,
+    gamma_i=3,
+    ion="p+",
+    m_e: u.kg = m_e,
+    m_i: u.kg = m_p,
+    theta: u.deg = 45 * u.deg,
+    z=1,
+):
```

Comment:
```suggestion
@validate_quantities(
    B={"can_be_negative": False},
    n_i={"can_be_negative": False},
    T_e={"can_be_negative": False, "equivalencies": u.temperature_energy()},
    T_i={"can_be_negative": False, "equivalencies": u.temperature_energy()},
)
def two_fluid_dispersion_solution(
    *,
    B: u.T,
    ion: Union[str, Particle],
    k: u.rad / u.m,
    n_i: u.m ** -3,
    T_e: u.K,
    T_i: u.K,
    gamma_e: Union[float, int] = 1,
    gamma_i: Union[float, int] = 3,
    theta: u.deg = 45 * u.deg,
    z_mean: Union[float, int] = None,
):
```

Here I've...

1. removed the `m_e` and `m_i` arguments since they were never used in the function
2. made `ion` a required argument (this as been a recent decision to do across the pkg...see issue #453 and PR #911)
3. require `k` to have units `rad/m` to be consistent with it actual units
4. added some typing annotations
5. renamed `z` -> `z_mean` and created default value of `None` (this is more consistent with the `z_mean` usage across the pkg)
6. removed from `@validation_quantities` the validation for `theta` since the default behavior is `can_be_negative == True`